homeassistant/components/renson/button.py

```diff
@@ -0,0 +1,73 @@
+"""Renson ventilation unit buttons."""
+from __future__ import annotations
+
+from _collections_abc import Callable
+from renson_endura_delta.renson import RensonVentilation
+
+from homeassistant.components.button import (
+    ButtonDeviceClass,
+    ButtonEntity,
+    ButtonEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import RensonCoordinator, RensonData
+from .const import DOMAIN
+from .entity import RensonEntity
+
+SYNC_TIME_BUTTON: ButtonEntityDescription = ButtonEntityDescription(
+    key="sync_time",
+    entity_category=EntityCategory.CONFIG,
+    translation_key="sync_time",
+)
+
+RESTART_BUTTON: ButtonEntityDescription = ButtonEntityDescription(
+    key="restart",
+    device_class=ButtonDeviceClass.RESTART,
+    entity_category=EntityCategory.CONFIG,
+)
```

Comment:
```suggestion
@dataclass
class RensonButtonEntityDescriptionMixin:
    """Action function called on press."""

    action_fn: Callable[[RensonVentilation], None]


@dataclass
class RensonButtonEntityDescription(
    ButtonEntityDescription, RensonButtonEntityDescriptionMixin
):
    """Class describing Renson button entity."""


ENTITY_DESCRIPTIONS: tuple[RensonButtonEntityDescription, ...] = (
    RensonButtonEntityDescription(
        key="sync_time",
        entity_category=EntityCategory.CONFIG,
        translation_key="sync_time",
        action_fn=lambda api: api.sync_time(),
    ),
    RensonButtonEntityDescription(
        key="restart",
        device_class=ButtonDeviceClass.RESTART,
        entity_category=EntityCategory.CONFIG,
        action_fn=lambda api: api.restart_device(),
    ),
)
```

This improves maintainability as adding new buttons in the feature is easy and we are trying to get consistency over all integrations.